src/proxy.c

```diff
@@ -33,11 +33,22 @@ SEXP vec_proxy_equal(SEXP x) {
   return vec_proxy_recursive(x, vctrs_proxy_equal);
 }
 SEXP vec_proxy_equal_dispatch(SEXP x) {
-  if (vec_typeof(x) == vctrs_type_s3) {
-    return vctrs_dispatch1(syms_vec_proxy_equal_dispatch, fns_vec_proxy_equal_dispatch,
-                           syms_x, x);
-  } else {
-    return x;
+  switch (vec_typeof(x)){
```

Comment:
It may be easier to unwrap the result of `vec_proxy_recursive()` in `vec_proxy_equal()` instead? Then you don't need to check for S3 etc.

Reply:
Thanks. I updated it to call a new `vec_proxy_unwrap()` function.